### PR TITLE
Code Improvement: sam deploy calls

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)

## Description

Improving the code that calls `sam deploy`. This change will be used in a follow up PR to further improve the sam deploy workflow code.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/sam/cli/samCliDeploy.ts [hunk 1]
```diff
@@ -8,55 +8,53 @@
...
-        const logger: Logger = getLogger()
-
-        const args = [
-            'deploy',
-            '--template-file', this.templateFile,
-            '--stack-name', this.stackName,
-            '--capabilities', 'CAPABILITY_IAM',
-            '--region', this.region,
-            '--profile', this.profile
+export interface SamCliDeployArgs {
```
Comment (mpiroc) [inline src/shared/sam/cli/samCliDeploy.ts]: Re offline discussion about consistent naming: `SamCliDeployParameters`.

File: src/test/shared/sam/cli/samCliTestUtils.ts [hunk 2]
```diff
@@ -26,3 +27,36 @@ export class MockSamCliProcessInvoker implements SamCliProcessInvoker {
...
     }
 }
+
+export function assertArgsContainArgument(
+    args: any[],
+    argOfInterest: string,
+    expectedArgValue: string
+) {
+    const argPos = args.indexOf(argOfInterest)
+    assert.notStrictEqual(argPos, -1, `Expected arg ${argOfInterest} was not found`)
```
Comment (mpiroc) [inline src/test/shared/sam/cli/samCliTestUtils.ts]: Nit: Throughout these messages what do you think about wrapping `argOfInterest` in quotes, i.e. `'${argOfInterest}'`?

File: src/shared/sam/cli/samCliDeploy.ts [hunk 3]
```diff
@@ -8,55 +8,53 @@
...
-        console.error(`stdout: ${stdout}`)
-
-        const message = error && error.message ? error.message : stderr || stdout
-        const err = new Error(`sam deploy encountered an error: ${message}`)
-        logger.error(err)
-        throw err
+        args.push('--parameter-overrides', ...overrides)
     }
+
+    const { exitCode, error, stderr, stdout }: ChildProcessResult = await invoker.invoke(...args)
```
Comment (mpiroc) [inline src/shared/sam/cli/samCliDeploy.ts]: We should also movie this boilerplate error checking to a common location. Can be in a separate PR if it's out-of-scope for this PR.

Comment (awschristou) [reply src/shared/sam/cli/samCliDeploy.ts]: It's like you're reading my mind! 
After this PR I am making a similar change to sam package, and then I planned to unify the error checking in a separate PR (to avoid PR collisions and bigger PRs)

File: src/test/shared/sam/cli/samCliTestUtils.ts (hunk 2 above)
Comment (awschristou) [reply src/test/shared/sam/cli/samCliTestUtils.ts]: Good utility in that, especially because we're using these methods a fair bit

File: src/lambda/commands/deploySamApplication.ts [hunk 4]
```diff
@@ -166,22 +166,24 @@ async function buildPackageDeploy(params: {
...
+
+        await runSamCliDeploy(
+            {
+                parameterOverrides: params.deployParameters.parameterOverrides,
+                profile: params.deployParameters.profile,
+                templateFile: packageTemplatePath,
+                region: params.deployParameters.region,
+                stackName: params.deployParameters.destinationStackName,
+            },
+            params.invoker,
```
Comment (bryceitoc9) [inline src/lambda/commands/deploySamApplication.ts]: Pass in `channelLogger.logger`
